task-maker-format/src/ioi/sanity_checks/task.rs

```diff
@@ -121,3 +125,95 @@ impl SanityCheck for NoBitsStdCpp {
         Ok(())
     }
 }
+
+/// Check that "implied" subtasks dependecies do not form cycles.
+#[derive(Debug, Default)]
+pub struct SubtaskDependencies;
+make_sanity_check!(SubtaskDependencies);
+
+impl SanityCheck for SubtaskDependencies {
+    type Task = IOITask;
+
+    fn name(&self) -> &'static str {
+        "SubtaskDependencies"
+    }
+
+    fn category(&self) -> SanityCheckCategory {
+        SanityCheckCategory::Task
+    }
+
+    fn post_hook(&self, task: &Self::Task, eval: &mut EvaluationData) -> Result<(), Error> {
```

Comment:
Is there a reason why you are using a `post_hook` instead of a `pre_hook`?

Reply:
It's now a pre_hook